core/src/main/scala/kafka/tools/ConsoleProducer.scala

```diff
@@ -110,6 +110,10 @@ object ConsoleProducer {
       props, ProducerConfig.SEND_BUFFER_CONFIG, config.options, config.socketBufferSizeOpt)
     CommandLineUtils.maybeMergeOptions(
       props, ProducerConfig.BUFFER_MEMORY_CONFIG, config.options, config.maxMemoryBytesOpt)
+    CommandLineUtils.maybeMergeOptions(
```

Comment:
We can add a comment above this line, to mention why we set batch.size with two different option values. Ex:
// We currently have 2 options to set the batch.size value. We'll deprecate/remove one of them in KIP-717.